packages/beacon-node/src/network/processor/index.ts

```diff
@@ -0,0 +1,161 @@
+import {Logger, mapValues} from "@lodestar/utils";
+import {IBeaconChain} from "../../chain/interface.js";
+import {Metrics} from "../../metrics/metrics.js";
+import {NetworkEvent, NetworkEventBus} from "../events.js";
+import {GossipType} from "../gossip/interface.js";
+import {createGossipQueues} from "./gossipQueues.js";
+import {NetworkWorker, NetworkWorkerModules} from "./worker.js";
+import {PendingGossipsubMessage} from "./types.js";
+import {ValidatorFnsModules, GossipHandlerOpts} from "./gossipHandlers.js";
+
+export type NetworkProcessorModules = NetworkWorkerModules &
+  ValidatorFnsModules & {
+    chain: IBeaconChain;
+    events: NetworkEventBus;
+    logger: Logger;
+    metrics: Metrics | null;
+  };
+
+export type NetworkProcessorOpts = GossipHandlerOpts & {
+  maxGossipTopicConcurrency?: number;
+};
+
+const executeGossipWorkOrderObj: Record<GossipType, true> = {
```

Comment:
might be worth comment here mentioning the order here defines the order that gossip topics will be processed